From 8322e4157bf49d992b5b9e460f2c0785865dd1c1 Mon Sep 17 00:00:00 2001
From: Yashpal Dutta <yashpal.dutta@freescale.com>
Date: Thu, 20 Mar 2014 19:55:51 -0500
Subject: [PATCH 07/26] Fixed private key support for DH

Upstream-status: Pending

Required Length of the DH result is not returned in dh method in openssl

Tested-by: Yashpal Dutta <yashpal.dutta@freescale.com>
---
 crypto/dh/dh_ameth.c | 7 -------
 1 file changed, 7 deletions(-)

diff --git a/crypto/dh/dh_ameth.c b/crypto/dh/dh_ameth.c
index ed32004..02ec2d4 100644
--- a/crypto/dh/dh_ameth.c
+++ b/crypto/dh/dh_ameth.c
@@ -422,13 +422,6 @@ static int dh_copy_parameters(EVP_PKEY *to, const EVP_PKEY *from)
 	if (to->pkey.dh->g != NULL)
 		BN_free(to->pkey.dh->g);
 	to->pkey.dh->g=a;
-	if ((a=BN_dup(from->pkey.dh->q)) != NULL) {
-		if (to->pkey.dh->q != NULL)
-			BN_free(to->pkey.dh->q);
-		to->pkey.dh->q=a;
-	}
-
-	to->pkey.dh->length = from->pkey.dh->length;
 
 	return 1;
 	}
-- 
2.3.5

